### PR TITLE
Initial AArch64 support for live media

### LINKED ISSUE #160: livecd-tools cannot create aarch64 media
Are threr any plan to support aarch64 ?

### PR DESCRIPTION
**Pull request by Conan-Kudo**
Refactored version of #180 

Obsoletes #180

Fixes #160

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@psgreco, @FGrose, @fedya: Please take a look and test this please? I had to do some initial adapting from the original PR to make it work on Fedora and CentOS.

**Comment by Conan-Kudo** [comment]
@fedya: I think this should also largely work on OpenMandriva, since they also ship GRUB EFI binaries in RPMs in `/boot`.

**Comment by Conan-Kudo** [comment]
@martinezjavier Can you please take a look at this to see if I'm doing it right?

**Comment by fedya** [comment]
working fine

File: imgcreate/live.py [hunk 1]
```diff
@@ -1030,14 +1030,186 @@ def _get_excluded_packages(self):
...
+    def _get_xorrisofs_options(self, isodir):
+        options = []
+        if os.path.exists(os.path.join(isodir, "images/efiboot.img")):
+            options += ["-eltorito-alt-boot", "-e", "images/efiboot.img",
+                        "-no-emul-boot", "-hide-rr-moved"]
+        options += ["-rational-rock", "-joliet", "-volid", self.fslabel]
+        return options
+
+    def _get_required_packages(self):
+        return ["grub2-efi"] \
```
Comment (martinezjavier) [inline imgcreate/live.py:1048]: I think it would be better for this to be `grub2-efi-aa64`.

File: imgcreate/live.py [hunk 2]
```diff
@@ -1030,14 +1030,186 @@ def _get_excluded_packages(self):
...
+    def efiarch(self):
+        if not self._efiarch:
+            # for most things, we want them named boot$efiarch
+            efiarch = {"aarch64": "AA64"}
+            self._efiarch = efiarch[dnf.rpm.basearch(hawkey.detect_arch())]
+        return self._efiarch
+
+    def __copy_efi_files(self, isodir):
+        fail = False
+        files = [("/boot/efi/EFI/*/shim.efi", "/EFI/BOOT/BOOT%s.EFI" % (self.efiarch,), True),
```
Comment (martinezjavier) [inline imgcreate/live.py:1101]: `shim.efi` is an artifact from where there wasn't support in Fedora for `ia32` and is still installed by the shim package for backward compatibility. You need to use `shimaa64.efi` instead.

I noticed that you copied this from the `x86LiveImageCreator` class, so probably needs to be changed there too.

In fact, couldn't you just use `self.efiarch` everywhere and have a base class that factors out all this common code ?

File: imgcreate/live.py [hunk 3]
```diff
@@ -1030,14 +1030,186 @@ def _get_excluded_packages(self):
...
     LiveImageCreator = x86LiveImageCreator
 elif arch in ("ppc",):
     LiveImageCreator = ppcLiveImageCreator
 elif arch in ("ppc64",):
     LiveImageCreator = ppc64LiveImageCreator
-elif arch.startswith(("arm", "aarch64")):
+elif arch.startswith(("aarch64")):
+    LiveImageCreator = aarch64LiveImageCreator
+elif arch.startswith(("arm")):
     LiveImageCreator = LiveImageCreatorBase
```
Comment (martinezjavier) [inline imgcreate/live.py:1213]: What's the effect of setting `LiveImageCreator` to `LiveImageCreatorBase` ? By looking at that class it seems that image won't work for these arches. Do you want to add an `armLiveImageCreator` that's similar to `aarch64LiveImageCreator` ? Maybe as mentioned you could just create a generic `efiLiveImageCreator` or something and just set `self.efiarch`.

Comment (Conan-Kudo) [reply imgcreate/live.py:1213]: Setting it to `LiveImageCreatorBase` means it doesn't work and needs to be implemented.

**Pull request merged**